sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2390,13 +2345,13 @@ static pi_result SetKernelParamsAndLaunch(
     const KernelArgMask *EliminatedArgMask,
     const std::function<void *(Requirement *Req)> &getMemAllocationFunc,
     bool IsCooperative) {
+  assert(Queue && "Queue with submitted kernel could not be on host");
```

Comment:
May be this is better? Also applies for several other assertions in this file.
```suggestion
  assert(Queue && "Kernel submissions should have an associated queue");
```
The "device queue must..." wording that appears later on works too.